src/geometamaker/config.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import os
+
+import platformdirs
+
+from . import models
+
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CONFIG_PATH = os.path.join(
+    platformdirs.user_config_dir(), 'geometamaker_profile.yml')
+
+
+class Config(object):
+    """Encapsulates user-settings such as a metadata Profile."""
+
+    def __init__(self, config_path=DEFAULT_CONFIG_PATH):
+        """Load a Profile from a config file.
+
+        Use a default user profile if none given.
+
+        Args:
+            config_path (str): path to a local yaml file
+
+        """
+        self.config_path = config_path
+        self.profile = models.Profile()
+
+        try:
+            self.profile = models.Profile.load(self.config_path)
+        except FileNotFoundError as err:
+            LOGGER.debug(err)
+            pass
+        # an invalid profile should raise a TypeError
+        except TypeError as err:
+            LOGGER.warning(err)
+            LOGGER.warning(
+                f'{self.config_path} contains an inavlid profile. '
+                'It will be ignored. You may wish to delete() it.')
```

Comment:
In either of these exception handlers, is the full traceback important?  It does seem like it'll be lost here, but could easily be preserved (if it's important) with a `LOGGER.exception` or else by using the `traceback` module to fetch the formatted traceback string.

Reply:
Thanks, yes I do think it's worth logging the full traceback in the case where we're also issuing the `warning`. In the `FileNotFoundError`, I don't think we want to disturb the user with `ERROR`-level logging, so I discovered the `exc_info` arg of `debug()`